File: packages/app/android/src/flutter/java/io/invertase/firebase/app/FlutterFirebaseAppPlugin.java
package io.invertase.firebase.app;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry;

import static io.invertase.firebase.common.FlutterFirebasePlugin.getChannelName;

public class FlutterFirebaseAppPlugin implements MethodCallHandler {
  public static void registerWith(PluginRegistry.Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), getChannelName("app"));
//    channel.setMethodCallHandler(new FlutterFirebaseAnalyticsPlugin(registrar));
  }

  @Override
  public void onMethodCall(MethodCall methodCall, Result result) {

  }
}

File: packages/app/android/src/flutter/java/io/invertase/firebase/app/FlutterFirebaseAppPlugin.kt
package io.invertase.firebase.app

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.plugin.common.PluginRegistry.Registrar

class FlutterFirebaseAppPlugin : MethodCallHandler {

  override fun onMethodCall(methodCall: MethodCall, result: Result) {

  }

  companion object {
    @JvmStatic
    fun registerWith(registrar: Registrar) {
      MethodChannel(registrar.messenger(), "analytics")
      // TODO set channel
    }
  }
}
